File: web_deprecated/src/main/java/dev/recing/web/fetch/RecipeFetchException.java
package dev.recing.web.fetch;

public class RecipeFetchException extends RuntimeException {

    private final FetchErrorCode code;

    public RecipeFetchException(FetchErrorCode code, Object... args) {
        super(code.getMessage(args));
        this.code = code;
    }

    public FetchErrorCode getCode() {
        return code;
    }
}

File: packages/ingestion/src/recipe-fetch-exception.ts
/**
 * Error thrown when a URL fetch operation fails.
 * Carries the machine-readable error code and a human-friendly message.
 */

import { FETCH_ERROR_MESSAGES, FetchErrorCode, resolveErrorMessage } from "@recing/schema";

export class RecipeFetchException extends Error {
  constructor(
    public readonly code: FetchErrorCode,
    ...args: unknown[]
  ) {
    super(resolveErrorMessage(FETCH_ERROR_MESSAGES, code, ...args));
    this.name = this.constructor.name;
  }
}
